File: 프로그래머스/Lv.1/유연근무제.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> schedules, vector<vector<int>> timelogs, int startday) {
    int answer = 0;

    auto toMinutes = [](int hhmm) {
        return (hhmm / 100) * 60 + (hhmm % 100);
    }; 

    for (int i = 0; i < timelogs.size(); i++) {
        int deadline = toMinutes(schedules[i]) + 10;
        int cnt = 0;

        for (int j = 0; j < timelogs[i].size(); j++) {
            int dow = (startday + j) % 7;
            if (dow == 6 || dow == 0)
                continue;

            int curTime = toMinutes(timelogs[i][j]);
            if (0 <= curTime && curTime <= deadline) {
                cnt++;
            }
        }

        if (cnt == 5)
            answer++;
    }

    return answer;
}

File: 프로그래머스/lv1/388351. 유연근무제/유연근무제.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] schedules, int[,] timelogs, int startday) {
        int answer = 0;
        
        int people = timelogs.GetLength(0);
        int day = timelogs.GetLength(1);
        
        for(int i = 0; i < people; ++i) {
            bool isOK = true;
            int curday = startday;
            
            int scheduletime = (schedules[i] / 100) * 60 + (schedules[i] % 100); 
            
            for(int j = 0; j < day; ++j) {
                int curtime = (timelogs[i, j] / 100) * 60 + (timelogs[i, j] % 100); 
                int diff = curtime - scheduletime;
                
                if(diff > 10 && curday != 6 && curday != 7) {
                    isOK = false;
                    break;
                }

                curday++;
                if(curday > 7) curday -= 7;
                
            }
            if(isOK)
                answer++;
        }
        
        return answer;
    }
}